### ISSUE TITLE
[Android Bootcamp] Project 3: CPTwitterClient - Ready for Review

### ISSUE THREAD

**Issue by Vikramjeet-Singh** [Vikramjeet-Singh/CPTwitterClient#1]
My app CPTwitterClient is complete. Please review. /cc @codepathreview @codepath

P.S. - I was not able to complete following stories because I was sick most of the week:
1. Advanced: User can open the twitter app offline and see last loaded tweets
                      Tweets are persisted into sqlite and can be displayed from the local DB
2. Advanced: Improve the user interface and theme the app to feel "twitter branded"
3. Bonus: User can see embedded image media within the tweet detail view

Would that be ok if I try to finish these stories along with next week's project?

**Comment by nesquena** [comment]
> Would that be ok if I try to finish these stories along with next week's project?

Of course, you definitely should. Next week's project is building on top of the twitter client you started this week so it's a great opportunity to add several of the week 3 optionals in addition to the 4th assignment we will introduce tomorrow. Full review shortly.

**Comment by codepathreview** [comment]
:+1: Great work! A few notes after checking out the code:
- Good to see you were able to format the timestamp cleanly (consider externalizing to utils class)
- Nice to see you used the `SwipeRefreshLayout` for easy timeline refreshes
- Good to see you refreshed the timeline after composing a new tweet
- Great job implementing the compose tweet feature as a popup activity
- Consider adding local persistence of tweets by using ActiveAndroid in the `Tweet` and `User`. See the [persistence guide](http://guides.codepath.com/android/Persisting-Data-to-the-Device#object-relational-mappers) and this [other guide](http://guides.codepath.com/android/ActiveAndroid-Guide) for more details.
-  Nice touch with the character count being displayed while the user composes a tweet 

Here's a detailed [Project 3 Feedback Guide here](http://courses.codepath.com/snippets/intro_to_android/project_3_feedback) which covers the most common issues with this submitted project. Read through the feedback guide point-by-point to determine how you could improve your submission. 

Let us know if you have any other thoughts or questions about this assignment. Hopefully by now you feel pretty comfortable with all the major pieces to basic Android apps (Views, Controllers, ActionBar, Navigation, Models, Authentication, API Communication, Persistence, et al) and see how they all fit together. We are close now to a turning point in the course where you should be hitting a "critical mass" towards your knowledge of Android.
